curator-framework/src/main/java/org/apache/curator/framework/api/CreateBuilderMain.java

```diff
@@ -27,6 +29,56 @@
     Compressible<CreateBackgroundModeStatACLable>,
     Statable<CreateProtectACLCreateModePathAndBytesable<String>>
 {
+
+    /**
+     * First 3 characters in the prefix on ZNode names when using {@link #withProtection()}
+     */
+    static final String PROTECTED_PREFIX = "_c_";
+
+    /**
+     * Last character used in the prefix on ZNode names when using {@link #withProtection()}
+     */
+    static final char PROTECTED_SEPARATOR = '-';
+
+    /**
+     * Prefix length on ZNode name when using {@link #withProtection()}
+     */
+    static final int PROTECTED_PREFIX_WITH_UUID_LENGTH = 
+            PROTECTED_PREFIX.length() 
+            + 36 // UUID canonical text representation produced by {@link UUID#toString()}
+            + 1; // Separator length
+
+    /**
+     * Utility method to determine if a given ZNode name starts with Curator's generated protected prefix.
+     * 
+     * @param znodeName ZNode name
+     * @return {@code true} if the given ZNode name starts with Curator's generated protected prefix
+     */
+    public static boolean isProtectedZNode(final String znodeName) {
+        if (znodeName.length() > PROTECTED_PREFIX_WITH_UUID_LENGTH
+                && znodeName.startsWith(PROTECTED_PREFIX)
+                && znodeName.charAt(PROTECTED_PREFIX_WITH_UUID_LENGTH-1) == PROTECTED_SEPARATOR
+           ) {
+            try {
+                UUID.fromString(znodeName.substring(PROTECTED_PREFIX.length(), PROTECTED_PREFIX_WITH_UUID_LENGTH-2));
+                return true;
+            } catch (IllegalArgumentException e) {
+                // Just false
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Utility method to remove Curator's generated protected prefix if present
+     * 
+     * @param znodeName ZNode name
+     * @return string without Curator's generated protected prefix if present; original string if prefix not present
+     */
+    public static String removeProtectedPrefix(final String znodeName) {
```

Comment:
What if returns `Optional<String>` which equals `isProtectedZNode(znodeName) ? Optional.of(znodeName.substring(PROTECTED_PREFIX_WITH_UUID_LENGTH)) : Optional.empty()`.

For API semantic it fits in "if present" and user can easily get the original sting by `.orElse` but hard to filter non-protected znodes with this API(the user of this method seems interested in protected znodes only.)

Reply:
The purpose it is just the opposite and not to worry if the ZNode is protected or not when using the API. Why is hard to filter non-protected znodes? You pass the ZNode name and if the ZNode is not protected you get the same.
Basically, this method provides you a way to normalize the names and remove the non-semantic part of it (does not matter if the node is protected or not)
I do not think that Optional is a good fit.  The method provides a mean  to get the normalized path (without prefix) given a real path, so it can never be null. All valid ZNodes names will have a normalized form without the prefix.

Reply:
Thanks for your clarification. It makes sense we add a method for normalizing which works as you described. What about name it `normalize` and document the behavior as you said.

Normalizing is the purpose and remove prefix if it is a protected node is the implementation.

Reply:
Yeah - let's leave it as it.